pkg.nix

```diff
@@ -0,0 +1,173 @@
+# NOTE: This file is supposed to be similar to what is in Nixpkgs, except for
+# the `version`, `src` and `libtexpdf-src` attributes that are given by the
+# `flake.nix`. In Nixpkgs, we don't need `libtexpdf-src` because we use
+# `fetchFromGitHub` with fetchSubmodules = true;`.
+{ lib
+, stdenv
+, version, src, libtexpdf-src
+, autoreconfHook
+, gitMinimal
+, pkg-config
+, jq
+, cargo
+, rustc
+, rustPlatform
+, makeWrapper
+, poppler_utils
+, harfbuzz
+, icu
+, fontconfig
+, lua
+, libiconv
+, darwin
+, makeFontsConf
+, gentium
+, runCommand
+}:
+
+let
+  luaEnv = lua.withPackages(ps: with ps; [
```

Comment:
I always wondered, what is the lua version you feel most comfortable with? Nixpkgs' `lua` is `lua5.2`.

Reply:
SILE is 100% compatible with all versions of Lua *newer* than 5.0 (currently 5.1–5.4). Up through v0.14.6, we were recommending the distro default `lua` as the path of least resistance. For Ubuntu LTS and some BSDs that has been 5.2, for almost everybody else that is 5.4. Starting with [v0.14.7](https://github.com/sile-typesetter/sile/releases/tag/v0.14.7) we also have 100% feature parity on LuaJIT. Previously we supported it but debug features were limited; now it is no sweat off anybody's back to use. Since LuaJIT is handily 2× faster we now recommend it everywhere that happens to have distro support for it. At some point it will be the build time default, for now `./configure --with-luajit` is needed to switch, including relevant to this PR making that the VM we build into the Rust binary.

Reply:
I added 3 more nix packages (see `nix search .`) of builds of sile with different lua variants.

Reply:
Currently Nixpkgs' default `lua` which is 5.2, is used for the default build - by default I mean `defaultPackage = sile;`, and this is the build you get when you (will) run `nix run github:sile-typesetter/sile`.